src/app/maintenance-worker.ts

```diff
@@ -47,21 +47,26 @@ export class MaintenanceWorker implements IRunnable {
     for (const invoice of invoices) {
       debug('invoice %s: %o', invoice.id, invoice)
       try {
-        debug('getting invoice %s from payment processor', invoice.id)
+        debug('getting invoice %s from payment processor: %o', invoice.id, invoice)
         const updatedInvoice = await this.paymentsService.getInvoiceFromPaymentsProcessor(invoice)
         await delay()
-        debug('updating invoice status %s: %o', invoice.id, invoice)
+        debug('updating invoice status %s: %o', updatedInvoice.id, updatedInvoice)
         await this.paymentsService.updateInvoiceStatus(updatedInvoice)
 
         if (
           invoice.status !== updatedInvoice.status
           && updatedInvoice.status == InvoiceStatus.COMPLETED
-          && invoice.confirmedAt
+          && updatedInvoice.confirmedAt
         ) {
           debug('confirming invoice %s & notifying %s', invoice.id, invoice.pubkey)
+          const update = {
+              ...invoice,
+              ...{ amountPaid: invoice.amountRequested },
+              ...updatedInvoice,
```

Comment:
Should we use a [`deepMerge`](https://ramdajs.com/docs/#mergeDeepLeft) here?

Reply:
done